charts/loki-stack/templates/datasources.yaml

```
@@ -21,6 +21,9 @@ data:
       url: {{ tpl .Values.loki.url . | quote }}
       version: 1
       isDefault: {{ default false .Values.loki.isDefault }}
+      {{- if .Values.loki.datasource.uid }}
+      uid: {{ .Values.loki.datasource.uid | quote }}
+      {{- end }}
```

How about:

```suggestion
      {{- with .Values.loki.datasource.uid }}
      uid: {{ . | quote }}
      {{- end }}
```

If uid is empty, it will just get omitted.

---

good idea!